nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/StandardFlowService.java

```diff
@@ -284,7 +251,7 @@ public boolean isRunning() {
 
     @Override
     public void start() throws LifeCycleStartException {
-        writeLock.lock();
+       dao.writeProtectFlow();
```

Comment:
Thanks @joewitt, I'll move responsibility back over to the caller, along with your other feedback, and push an update.